Hardware/Hardware/CardReader/StripeCardReader/StripeCardReaderService.swift

```diff
@@ -207,6 +207,29 @@ extension StripeCardReaderService: CardReaderService {
         }
     }
 
+    public func waitForInsertedCardToBeRemoved() -> Future<Void, Error> {
+        return Future() { [weak self] promise in
+            guard let self = self else {
+                return
+            }
+
+            // If there is no chip card inserted, it is ok to immediately return. The payment method may have been swipe or tap.
+            if !self.isChipCardInserted {
+                return promise(.success(()))
+            }
+
+            self.timerCancellable = Timer.publish(every: 1, tolerance: 0.1, on: .main, in: .default)
```

Comment:
Nice use of Timer Publisher instead of `Foundation` Timers! 👏

Reply:
Kudos to Allen on this https://github.com/woocommerce/woocommerce-ios/commit/55807d8791e9a71f2f398e332a6d65d6e64c2e8f 👏